lib/util.js

```diff
@@ -230,17 +230,30 @@ function format(f) {
   return str;
 }
 
-var debugs = {};
-var debugEnviron;
+const debugs = {};
+let debugEnviron;
+
+const regExpSpecialChars = /[|\\{}()[\]^$+?.]/g;
+
+function arrayToRegExp(strArr) {
+  // Escape special chars except wildcard.
+  strArr = strArr.map((str) => {
+    return str.replace(regExpSpecialChars, '\\$&')
+      .replace(/\*/g, '.*');
+  });
+
+  return new RegExp(`^${strArr.join('|')}$`, 'i');
+}
 
 function debuglog(set) {
   if (debugEnviron === undefined) {
-    debugEnviron = new Set(
-      (process.env.NODE_DEBUG || '').split(',').map((s) => s.toUpperCase()));
+    debugEnviron = Array.from(new Set(
+      (process.env.NODE_DEBUG || '').split(',').map((s) => s.toUpperCase())));
```

Comment:
This part is weird. First an Array is created, then the Set, then an Array from the Set.

Reply:
@BridgeAR This happens for deduplication, do you have other suggestions? Converting to a Set and back is usualy what I do too

Reply:
The only thing I can think of is improving readability by adding one more statement like,
```
debugEnviron = (process.env.NODE_DEBUG || '').split(',').map((s) => s.toUpperCase())
debugEnviron = Array.from(new Set());
```
But i wonder if it worth.

Reply:
wait, we can convert to uppercase before splitting so to remove the `map` operator.
```
debugEnviron = (process.env.NODE_DEBUG || '').toUpperCase().split(',')
debugEnviron = Array.from(new Set());
```

Reply:
I think it's just that we use `[...spread]` syntax more than `Array.from()`, which has the added advantage of being safer in case `Array` or `Array.from` is overridden. In any case I'd prefer the first variant for clarity.

Reply:
Thx @TimothyGu  Nice solution! Updated.

Reply:
@addaleax the current implementation was not about deduplication, see #13841. I guess it was chosen to make sure you do not have to lookup the entry in a array as that is slower. Depulication for a environment variable that is passed in by the user is definitely surprising to me. I would expect people not to include the same entry multiple times ;-)
As this is a one time thing and debugging is always slower, I will not block it from being included if you really think that is necessary (even though I do not see the point in it).

What I would like is to move the replacement to before the splitting. Running the RegExp once is a lot better than running it for each entry (even though there are probably very few).

I would also argue that the part being lazy does not make all that much sense here and I would rather move it out instead (it became lazy in a independent commit 22c68fdc1dae40f0ed9c71a02f66e5b2c6353691).

Reply:
In case the replacement is done right away, the splitting and joining could also be replaced by a `replace()`.

Reply:
Hi @BridgeAR . Thx for details, appreciate that! 
> What I would like is to move the replacement to before the splitting. Running the RegExp once is a lot better than running it for each entry (even though there are probably very few).

make sense, thx

> In case the replacement is done right away, the splitting and joining could also be replaced by a replace().

Yes, it could. Would this reduce the readability?